src/foundry/client-esm/applications/api/dialog.d.mts

```diff
@@ -115,28 +115,19 @@ declare class DialogV2<
    *          callback. If the dialog was dismissed, and rejectClose is false, the
    *          Promise resolves to null.
    */
-  static confirm<Options extends DeepPartial<DialogV2.WaitOptions>, YesReturn = true, NoReturn = false>(
-    config?: Options & {
-      /** Options to overwrite the default yes button configuration. */
-      yes?: InexactPartial<DialogV2.Button<YesReturn>>;
-
-      /** Options to overwrite the default no button configuration. */
-      no?: InexactPartial<DialogV2.Button<NoReturn>>;
-    },
-  ): Promise<YesReturn | NoReturn | InferButtonReturnTypes<Options> | InferDismissType<Options>>;
+  static confirm<YesReturn, NoReturn, Options extends DeepPartial<DialogV2.ConfirmConfig<YesReturn, NoReturn>>>(
+    config?: Options,
+  ): Promise<InferConfirmReturnTypes<Options> | InferButtonReturnTypes<Options> | InferDismissType<Options>>;
```

Comment:
I don't like having so many types here. You can keep each component piece as their own helper type but make them internal. I'd recommend exposing this as `ConfirmReturnType<Options>`